hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordScanner.java

```diff
@@ -152,7 +160,7 @@ public void scan() {
       // iterate over the paths
       logFormatReaderWrapper = new HoodieLogFormatReader(fs,
           logFilePaths.stream().map(logFile -> new HoodieLogFile(new Path(logFile))).collect(Collectors.toList()),
-          readerSchema, readBlocksLazily, reverseReader, bufferSize);
+          internalSchema != null ? null : readerSchema, readBlocksLazily, reverseReader, bufferSize);
```

Comment:
when internalSchema is not null， we should forbid  AbstractHoodieLogRecordScanner useing readerSchema(which is a avro Schema) to read avro record; since AbstractHoodieLogRecordScanner may not be able to read correctly data by using readerSchema.